tests/integration/security/reachability/mtls_test.go

```diff
@@ -0,0 +1,192 @@
+
+	pilot := pilot2.NewOrFail(t, ctx, pilot2.Config{})
+	appsInstance := apps.NewOrFail(ctx, t, apps.Config{Pilot: pilot})
+
+	aApp, _ := appsInstance.GetAppOrFail("a", t).(apps.KubeApp)
+	bApp, _ := appsInstance.GetAppOrFail("b", t).(apps.KubeApp)
+
+	headlessApp, _ := appsInstance.GetAppOrFail("headless", t).(apps.KubeApp)
+	// App without sidecar.
+	nakedApp, _ := appsInstance.GetAppOrFail("t", t).(apps.KubeApp)
+
+	testCases := []struct {
+		configFile  string
+		namespace   string
+		connections []connect.Connection
+	}{
+		{
+			configFile: "global-mtls-on.yaml",
+			connections: []connect.Connection{
+				{
+					From:            aApp,
+					To:              bApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: true,
+				},
+				{
+					From:            aApp,
+					To:              headlessApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: true,
+				},
+				{
+					From:            nakedApp,
+					To:              bApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: false,
+				},
+			},
+		},
+		{
+			configFile: "global-mtls-permissive.yaml",
+			connections: []connect.Connection{
+				{
+					From:            aApp,
+					To:              bApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: true,
+				},
+				{
+					From:            nakedApp,
+					To:              bApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: true,
+				},
+			},
+		},
+		{
+			configFile: "global-mtls-off.yaml",
+			connections: []connect.Connection{
+				{
+					From:            aApp,
+					To:              bApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: true,
+				},
+				{
+					From:            nakedApp,
+					To:              bApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: true,
+				},
+			},
+		},
+		{
+			configFile: "single-port-mtls-on.yaml",
+			namespace:  appsInstance.Namespace().Name(),
+			connections: []connect.Connection{
+				{
+					From:            aApp,
+					To:              bApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: false,
+				},
+				{
+					From:            nakedApp,
+					To:              bApp,
+					Port:            80,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: false,
+				},
+				{
+					From:            aApp,
+					To:              bApp,
+					Port:            90,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: true,
+				},
+				{
+					From:            nakedApp,
+					To:              bApp,
+					Port:            90,
+					Protocol:        apps.AppProtocolHTTP,
+					ExpectedSuccess: true,
+				},
+			},
+		},
+	}
+
+	for _, c := range testCases {
+		namespace := c.namespace
```

Comment:
You should run these all as sub-tests, so that it's clear which one is failing.

Reply:
Done: changed to run as sub-tests.